pkg/controller/gc_controller.go

```diff
@@ -115,6 +115,12 @@ func (c *gcController) cleanBackups() {
 			}
 
 			for _, volumeBackup := range backup.Status.VolumeBackups {
+				if c.snapshotService == nil {
+					glog.Warningf("Cannot garbage-collect snapshot %s for backup %s/%s because server is not configured with PersistentVolumeProvider",
```

Comment:
Let's put a check before the DeleteBackup call to see if the backup has any VolumeBackups. If it does and the snapshotService is nil, then log and skip deleting the backup.